src/main.py

```diff
@@ -0,0 +1,52 @@
+#!/usr/bin/env python
+import rospy
+from gennav.envs import PolygonEnv, ScanEnv
+from gennav.planners import RRT, RRG, PRM, PRMStar, PotentialField, InformedRRTstar
+from gennav.utils import RobotState
+from gennav.utils.geometry import Point
+from gennav.utils.samplers import UniformCircularSampler, UniformRectSampler
+from gennav_ros.commander import Commander
+from gennav_ros.controller import Controller
+
+planner_dict = {
+    "RRT": RRT,
+    "PRM": PRM,
+    "PRMStar": PRMStar,
+    "PotentialField": PotentialField,
+    "RRG": RRG,
+    "InformedRRTstar": InformedRRTstar,
+}
+env_dict = {"PolygonEnv": PolygonEnv, "ScanEnv": ScanEnv}
+
+
+def yaml_to_params():
+    param_list = rospy.get_param_names()
+    param_dict = {}
+    for param in param_list:
+        param = param[1:]
+        param_dict[param] = rospy.get_param(param)
+    return param_dict
+
+
+planner_name = rospy.get_param("planner_name")
+env_name = rospy.get_param("env_name")
+obstacles = rospy.get_param("obstacle_data")
+for planner_chk in planner_dict:
+    if planner_name == planner_chk:
+        planner = planner_dict[planner_name]
+        break
+for env_chk in env_dict:
+    if env_name == env_chk:
+        env = env_dict[env_name]
+        break
```

Comment:
You can directly check if an object is one of the keys of a dict using `key in dict.keys()` i think. No need to loop over all the keys

Reply:
Ya but this also assigns the value of the key to planner which is the used in the next set of lines

Reply:
```python
if planner_name in planner_dict.keys():
    planner = planner_dict[planner_name]
```

I mean this. Would to the same job right?

Reply:
Ohh yes,my bad

Reply:
I'll change it and commit tomorrow